### PR TITLE
Fix use of input slice without dedup active

### LINKED ISSUE #1182: RGBGFX can't use input slice without deduplication
Invoking RGBGFX with an input slice and _without_ deduplication fails `assert(remainingTiles == 0)` [here](https://github.com/gbdev/rgbds/blob/7a39e9e569783962d54265809d8ec654b91c318b/src/gfx/process.cpp#L784).

The issue is that the initial value of `remainingTiles` doesn't account for the reduced number of tiles due to the input slice being set. The tiles visitor does use the correct count, so the loop would exit "early", with `remainingTiles > 0`.

A simple fix is to just start with the number of tiles in the slice if one is set, otherwise default to use the whole image size as it currently does. This works, but otherwise seems like a low quality fix:

- `remainingTiles` seems to only be for dealing with the trim option? Doing more with it seems to be moving in the wrong direction.
- `remainingTiles` & `visitAsTiles` are both doing the 'how many tiles' calculation/logic which is not good and this duplication is seemingly to blame for the issue.
- The tile visitor/iterator should have a known length and could be trimmed (ended early) based on that -- which appears to be more or less what the 'optimised' (deduplicated) path does.
- If `visitAsTiles` accessing the options is OK (it currently does), it could just check trim as well.

### PR DESCRIPTION
**Pull request by ISSOtm**
Fixes #1182

cc @quinnyo: can you test if this fixes your original issue?

### CODE REVIEW

File: src/gfx/process.cpp [hunk 1]
```diff
@@ -780,7 +780,9 @@ static void outputTileData(Png const &png, DefaultInitVec<AttrmapEntry> const &a
 		fatal("Failed to open \"%s\": %s", output.c_str(options.output), strerror(errno));
 	}
 
-	uint64_t remainingTiles = (png.getWidth() / 8) * (png.getHeight() / 8);
+	uint16_t width = options.inputSlice.width ? options.inputSlice.width * 8 : png.getWidth();
+	uint16_t height = options.inputSlice.height ? options.inputSlice.height * 8 : png.getHeight();
+	uint64_t remainingTiles = (width / 8) * (height / 8);
```
Comment (Rangi42) [inline src/gfx/process.cpp:785]:
```suggestion
	uint16_t width = options.inputSlice.width ? options.inputSlice.width : png.getWidth() / 8;
	uint16_t height = options.inputSlice.height ? options.inputSlice.height : png.getHeight() / 8;
	uint64_t remainingTiles = width * height;
```

Comment (ISSOtm) [reply src/gfx/process.cpp:785]: How about calling the variables `widthTiles` and `heightTiles`, then?

Comment (Rangi42) [reply src/gfx/process.cpp:785]: Sure.

**Review by Rangi42 — APPROVED** [review]

**Comment by quinnyo** [comment]
Yep, it works! (This is what I meant by the "simple fix".)

**Comment by ISSOtm** [comment]
Okay, great! Let's merge this, then.

**Pull request merged**